### PR TITLE
feat: Change app theme to Material Components

### PR DESCRIPTION
**Pull request by liveHarshit**
Part of #1237

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Need to define colorSecondary and other stuff replacing colorAccent etc

**Comment by liveHarshit** [comment]
> Need to define colorSecondary and other stuff replacing colorAccent etc

Mean.. replace colorAccent to colorSecondary. 
.
Currently, all text field and buttons are also need to change with material components.

**Comment by iamareebjamal** [comment]
That happens automatically if you are using the material components non bridge theme

**Comment by iamareebjamal** [comment]
Updates?

**Comment by liveHarshit** [comment]
> Updates?

Sorry, I'm currently busy in Fossasia summit. I will update the PR after 19th.

**Comment by liveHarshit** [comment]
Should I use bride themes for material components or change the layout for the components later?

**Comment by iamareebjamal** [comment]
Don't use the bridge theme. Just change the default theme and colors and verify that everything works and looks the same

**Comment by iamareebjamal** [comment]
There'll be no need of changes in layout. It's automatically handled by the library

**Comment by liveHarshit** [comment]
> There'll be no need of changes in layout. It's automatically handled by the library

Automatically, filled text input layouts are there, which are looking odd. Need to change in outline layouts.

**Comment by iamareebjamal** [comment]
Update. This Pr is blocking others

**Comment by liveHarshit** [comment]
mio-design_assets_1hg4iTKzTMMgtJRx7bhaE2kSYR5BRYz1g_color-colorsystem-schemecreation-theme

@iamareebjamal Can I change app colors, And use them?

**Comment by liveHarshit** [comment]
Because material components need more colors.

**Comment by iamareebjamal** [comment]
Just need to define primary and secondary with variants. All others default are enough

**Comment by haroldadmin** [comment]
I can help solve this one. Sketch for Mac has a Material theme plugin that can help you figure out the values of these different colors. I'll post the required values. Let me build a quick version of the app theme in Sketch.

**Comment by haroldadmin** [comment]
@liveHarshit Try these values:

* Surface
  * colorSurface = FFFFFF
  * colorSurfaceLight = FFFFFF
  * colorSurfaceDark = F5F5F5

* Primary
  * colorPrimary = 673AB7
  * colorPrimaryVariant = 311B92

* Secondary
  * colorSecondary = 03A9F4
  * colorSecondaryVariant = 01579B

**Comment by liveHarshit** [comment]
@haroldadmin Thank you

**Comment by haroldadmin** [comment]
This is not the complete set. There are also text colour values, but they aren't necessary. Try these to check if they are enough to transition the app theme.

**Comment by liveHarshit** [comment]
Updated.

<p align="center">
 
 
</p>

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -1,27 +1,51 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="@style/Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="@style/Theme.MaterialComponents.Light.DarkActionBar.Bridge">
```
Comment (iamareebjamal) [inline app/src/main/res/values/styles.xml]: No need for Bridge as told before

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
         <!-- Customize your theme here. -->
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
+
+        <!-- New MaterialComponents attributes. -->
+        <item name="colorPrimaryVariant">@color/colorPrimaryVariant</item>
+        <item name="colorSecondaryVariant">@color/colorSecondaryVariant</item>
```
Comment (iamareebjamal) [inline app/src/main/res/values/styles.xml]: Add colorSecondary and remove everything below

File: app/src/main/res/values/styles.xml (hunk 1 above)
Comment (liveHarshit) [reply app/src/main/res/values/styles.xml]: Sorry, It was by mistake.

**Comment by liveHarshit** [comment]
Updated

**Pull request merged**